src/config/options.rs

```diff
@@ -493,10 +505,28 @@ pub enum StyleEdition {
     Edition2021,
     #[value = "2024"]
     #[doc_hint = "2024"]
+    #[unstable_variant]
     /// [Edition 2024]().
     Edition2024,
 }
 
+impl From<StyleEdition> for rustc_span::edition::Edition {
+    fn from(edition: StyleEdition) -> Self {
+        match edition {
+            StyleEdition::Edition2015 => Self::Edition2015,
+            StyleEdition::Edition2018 => Self::Edition2018,
+            StyleEdition::Edition2021 => Self::Edition2021,
+            StyleEdition::Edition2024 => Self::Edition2024,
+        }
+    }
+}
+
+impl PartialOrd for StyleEdition {
+    fn partial_cmp(&self, other: &StyleEdition) -> Option<std::cmp::Ordering> {
+        rustc_span::edition::Edition::partial_cmp(&(*self).into(), &(*other).into())
```

Comment:
Another decision point for us, albeit a hopefully minor and uncontroversial one. Technically for the purposes of the Style Edition, 2015 == 2018 == 2021, which isn't directly reflected here. However, that has absolutely zero effect on the formatting one way or the other, and I think reusing the compiler's edition comparison gives us more flexibility in the future.

That's because the style edition process _does_ allow for formatting prescriptions for syntax that didn't exist at the time a Style Edition was stabilized to be backported to stable Style Editions where the parser Edition _does_ support that syntax. E.g. imagine if new syntax is added in 2025 (after the 2024 Style Edition is cut and stabilized) and that syntax will parse in Edition 2021 and 2024, but not 2018 nor 2015; a scenario I think we can better handle at relevant sites for that construct using the standard ordering.